server/src/main/java/de/pnp/manager/component/character/Talent.java

```diff
@@ -45,7 +69,31 @@ public EPrimaryAttribute getThirdAttribute() {
     return thirdAttribute;
   }
 
+  /**
+   * The {@link EPrimaryAttribute primary attributes} for a talent check in correct order.
+   */
   public List<EPrimaryAttribute> getAttributes() {
     return List.of(getFirstAttribute(), getSecondAttribute(), getThirdAttribute());
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    Talent talent = (Talent) o;
+    return getName().equals(talent.getName()) && Objects.equals(getGroup(),
```

Comment:
the first comparison should also use Objects.equals, as getName() can theoretically return null